minor refactoring

BamBam-sys opened the pull request.

mrcoded reviewed (changes requested):
please use correct PR title and give detailed description of fix

BamBam-sys commented:
> please use correct PR title and give detailed description of fix

### i implemented the following fixes

1. fixed validation error appearing when user tries to resubmit login form after getting an error prompt when trying to login prior
2. implemented logic to render on the header only login button on signup page, and signup button on login page
3. properly handled server error to prevent code breaking
